remotespark/livyclientlib/command.py

```diff
@@ -17,11 +20,23 @@ def __ne__(self, other):
         return not self == other
 
     def execute(self, session):
-        session.wait_for_idle()
-        data = {"code": self.code}
-        response = session.http_client.post_statement(session.id, data)
-        statement_id = response['id']
-        return self._get_statement_output(session, statement_id)
+        self._spark_events.emit_statement_execution_start_event(session.guid, session.kind, session.id, self.guid)
+        statement_id = -1
+        try:
+            session.wait_for_idle()
+            data = {"code": self.code}
+            response = session.http_client.post_statement(session.id, data)
```

Comment:
There should be a test for when this post statement throws, that we are emitting the right end event.